core/runtime.rs

```diff
@@ -1972,10 +1973,13 @@ impl JsRuntime {
   // Send finished responses to JS
   fn resolve_async_ops(&mut self, cx: &mut Context) -> Result<(), Error> {
     let isolate = self.v8_isolate.as_mut().unwrap();
-
-    let js_recv_cb_handle = self.state.borrow().js_recv_cb.clone().unwrap();
-    let global_realm = self.state.borrow().global_realm.clone().unwrap();
-    let scope = &mut global_realm.handle_scope(isolate);
+    let scope = &mut self
```

Comment:
Would it be possible to set this up as eg. Option, and use that to do a lazy-init scope so if there are no pending tasks then a HandleScope doesn't get created?

Reply:
Good point

Reply:
Unfortuantely, lazy initialization of Option or LazyCell is not smartenough to know that its called _only once_...It would think that `self` has to be borrowed for each loop which is not possible in safe Rust.

Reply:
I think that should not be an issue, maybe, if you do the HandleScope init on line 2005, where self.state is already borrowed?